polar-core/src/vm.rs

```diff
@@ -2042,82 +2042,71 @@ impl PolarVirtualMachine {
         assert_eq!(args.len(), 2);
         let item = &args[0];
         let iterable = &args[1];
+        let item_is_ground = item.is_ground();
 
-        match (item.value(), iterable.value()) {
-            (_, Value::List(list)) if list.is_empty() => {
-                // Nothing is in an empty list.
-                self.backtrack()?;
```

Comment:
If anyone else is curious why this is no longer needed:

We used to iterate through the list, and push an alternative for every element of the list. Which meant that "no alternatives" just evaluated to true.

At some point we changed this to an iterator and `self.choose` which already handles the "no alternatives" case as meaning backtrack.

Cool that we can delete this code!